### PR TITLE
adding doc yaml, adding new test and splitting by endpoint, adding PO…

### PR DESCRIPTION
**Pull request by foohonpie**
…ST games endpoint

### CODE REVIEW

File: app/Http/Requests/PostGamesRequest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class PostGamesRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return auth('api')->guest();
```
Comment (mink) [inline app/Http/Requests/PostGamesRequest.php:16]: Are unauthenticated users allowed to submit a game?

File: tests/Feature/Games/PostGamesTest.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            'url' => 'not_a_url',
+            'short_description' => $input_game->long_description,
+            'long_description' => '',
+            'image_path' => $input_game->image_path,
+            'approved' => 'string',
+            'verified' => $input_game->verified,
+            'promoted' => $input_game->promoted,
+        ]);
+
+        $response->assertStatus(422);
```
Comment (bpkennedy) [inline tests/Feature/Games/PostGamesTest.php:58]: Interesting choice of 422 for this.  I'm reading more about the spec differences between 400 and 422 and I suppose it comes down to "malformed syntax" versus ok-formed and simply unprocessable.

For this, I would question since we are using PHP validation and types to say that url should be a `url` then aren't we in malformed territory and this a 400 is appropriate?

File: tests/Feature/Games/PostGamesTest.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
+<?php
```
Comment (bpkennedy) [inline tests/Feature/Games/PostGamesTest.php:1]: Do we also want to make a test that checks/prevents duplicates (since that's part of the validation of a create game POST request)?

**Review by bpkennedy — CHANGES_REQUESTED** [review]

File: tests/Feature/Games/PostGamesTest.php (hunk 2 above)
Comment (mink) [reply tests/Feature/Games/PostGamesTest.php:58]: Laravel automatically returns HTTP 422 for input validation errors that come from AJAX requests.

This request could be considered well-formed with semantically incorrect contents, though either 400 or 422 could be valid for this request.

File: app/Http/Requests/PostGamesRequest.php (hunk 1 above)
Comment (foohonpie) [reply app/Http/Requests/PostGamesRequest.php:16]: Woops, good catch!

Users and guests should both be allowed to submit.

File: tests/Feature/Users/GetAuthenticatedUserTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        ])->get('/user');
+        $response->assertStatus(200);
+    }
+
+    /**
+     * An unauthenticated user cannot access user data
+     *
+     * @return void
+     */
+    public function testAnUnauthetnicatedUserCannotAccessTheUserPage()
```
[the remark this answers is not in the document]
Comment (foohonpie) [reply tests/Feature/Users/GetAuthenticatedUserTest.php:41]: Eyes of steel!  That's maybe not the right phrase.  🤔

File: tests/Feature/Games/GetGamesTest.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?php
```
[the remark this answers is not in the document]
Comment (foohonpie) [reply tests/Feature/Games/GetGamesTest.php:1]: I only see it once.  There's GetGames plural and GetGame singular though, was that what you were seeing?

File: tests/Feature/Games/PostGamesTest.php (hunk 3 above)
Comment (foohonpie) [reply tests/Feature/Games/PostGamesTest.php:1]: Yeah, there are some additions to User tests I'd like to do too, plus some DRY optimizations.  I'll put details up in a ticket.

Comment (foohonpie) [reply tests/Feature/Games/PostGamesTest.php:1]: Yeah, there are some additions to User tests I'd like to do too, plus some DRY optimizations. I'll put details up in a follow-up ticket for test improvements.

**Review by bpkennedy — APPROVED** [review]
